scipy/stats/_qmc.py

```diff
@@ -2001,11 +2006,19 @@ def __init__(
 
         # sample to generate per iteration in the hypersphere around center
         self.ncandidates = ncandidates
+        
+        if u_bounds is None:
+            u_bounds = np.ones(d)
+        if l_bounds is None:
+            l_bounds = np.zeros(d)
+        self.l_bounds, self.u_bounds = _validate_bounds(
+            l_bounds=l_bounds, u_bounds=u_bounds, d=int(d)
+        )
 
         with np.errstate(divide='ignore'):
             self.cell_size = self.radius / np.sqrt(self.d)
             self.grid_size = (
-                np.ceil(np.ones(self.d) / self.cell_size)
+                np.ceil(self.u_bounds / self.cell_size)
```

Comment:
Ok but then how does that take into account the lower bounds?

Reply:
On line 2072 `ind_min = np.maximum(indices - n, self.l_bounds.astype(int))`, this is where the lower bounds is taken into account.

Reply:
But shouldn't the lower bound also affect the grid size?

Side note: please do not resolve conversations. Maintainers in SciPy typically like to close themselves at it helps us to keep track of our review.

Reply:
You are correct. I took a look at the algorithm from the original paper and made some updates to the code. Lower bounds should now be properly taken into account